### PR TITLE
feat: introduce new service name format to use actual client names for beacon/validator nodes, fixes #46

### LINKED ISSUE #46: Use alternative names for beacon/execution nodes
I would like to propose a change for `cl-client-0-beacon` name to become `teku-geth-0` and `el-client-0` to be called `geth-teku-0`. 
Its quite hard to keep track of which pair is what, when you run a lot of client mixes
Basically I would like to have <el>-<cl> for the EL node, and <cl>-<el> for the CL node

https://github.com/kurtosis-tech/eth-network-package/blob/07d81fa0b5e5a2d71598da15033c07f1a07068b7/src/participant_network.star#L156

### PR DESCRIPTION
**Pull request by leeederek**
Fixes #46: introduce a new service name format whereby we will now use a global index that starts at 1 and the name of the services will follow a cl_name-el_name / el_name-cl_name convention. 

As a concerete example, when given 2 nodes, one that is geth/lighthouse and the other being geth/teku:

***BEFORE***
cl-client-0-beacon
cl-client-0-validator  
cl-client-1 
el-client-0
el-client-1 

***AFTER***
1-geth-teku
1-teku-geth
2-geth-lighthouse
2-lighthouse-geth
2-lighthouse-geth-validator

This change was made to reflect and match the naming convention of the services that live in production for the Ethereum Foundation's dev/testnet

### CODE REVIEW

File: src/participant_network.star [hunk 1]
```diff
@@ -92,7 +89,8 @@ def launch_participant_network(plan, participants, network_params, global_log_le
 			fail("Unsupported launcher '{0}', need one of '{1}'".format(el_client_type, ",".join([el.name for el in el_launchers.keys()])))
 		
 		el_launcher, launch_method = el_launchers[el_client_type]["launcher"], el_launchers[el_client_type]["launch_method"]
-		el_service_name = "{0}{1}".format(EL_CLIENT_SERVICE_NAME_PREFIX, index)
+		el_service_name = "{0}-{1}-{2}".format(el_client_type, participant.cl_client_type, index)
```
Comment (barnabasbusa) [inline src/participant_network.star:92]: index should be client pair specific index, not global index.

**Review by h4ck3rk3y — COMMENTED** [review]
Can this behavior be put behind a flag?

**Comment by barnabasbusa** [comment]
Any reason you would want it behind a flag? What are the benefits of not knowing what client pair you looking at?

**Comment by h4ck3rk3y** [comment]
Was worried that some downstream implementations might rely on the previous service-names but ideally that isn't the case and we can catch/fix those

File: src/cl/lighthouse/lighthouse_launcher.star [hunk 2]
```diff
@@ -77,7 +76,7 @@ def launch(
 	extra_beacon_params,
 	extra_validator_params):
 
-	beacon_node_service_name = "{0}-{1}".format(service_name, BEACON_SUFFIX_SERVICE_NAME)
+	beacon_node_service_name = "{0}".format(service_name)
```
Comment (h4ck3rk3y) [inline src/cl/lighthouse/lighthouse_launcher.star:79]: can set this to just `beacon_node_service_name = service_name`

File: src/cl/lodestar/lodestar_launcher.star [hunk 3]
```diff
@@ -63,7 +62,7 @@ def launch(
 	extra_beacon_params,
 	extra_validator_params):
 
-	beacon_node_service_name = "{0}-{1}".format(service_name, BEACON_SUFFIX_SERVICE_NAME)
+	beacon_node_service_name = "{0}".format(service_name)
```
Comment (h4ck3rk3y) [inline src/cl/lodestar/lodestar_launcher.star:65]: can set this to just beacon_node_service_name = service_name

File: src/cl/lighthouse/lighthouse_launcher.star (hunk 2 above)
Comment (h4ck3rk3y) [reply src/cl/lighthouse/lighthouse_launcher.star:79]: is it clear that the non `-validator` node is beacon???

**Review by h4ck3rk3y — APPROVED** [review]

**Comment by barnabasbusa** [comment]
🥳

**Pull request merged**